packages/actionmenu/src/react/icon.tsx

```diff
@@ -11,12 +10,13 @@ const styles = ({ marginLeft, marginRight }) =>
     marginLeft && stylesheet[`.psds-actionmenu__icon-left`]
   )
 
-export const Icon = ({ marginLeft, marginRight, ...props }) => (
+interface Props extends HTMLAttributes<HTMLSpanElement> {
+  marginRight: boolean;
+  marginLeft: boolean;
+}
+
+export const Icon: React.FC<Props> = ({ marginLeft, marginRight, ...props }) => (
   <span {...filterReactProps(props)} {...styles({ marginLeft, marginRight })} />
 )
```

Comment:
```suggestion
export const Icon: React.FC<Props> = ({ marginLeft, marginRight, ...rest }) => (
  <span {...rest} {...styles({ marginLeft, marginRight })} />
)
```

Reply:
an example of using types to ensure only props that are supported are passed - and we filter out the props that are not supported on the el. we can use this pattern to remove our need for filterReactProps